server/server.js

```diff
@@ -40,13 +35,35 @@ app.post("/setPiece", async (req, res) => {
   }
 
   currentColumns[x] = y - 1; // so row moves up by 1 row
-  console.log("BOARD", boardArr);
   res.status(200);
   res.send();
+  console.log("CURRENT PLAYER", currentPlayer);
 });
 
 app.get("/fetchGameStateVariables", (req, res) => {
-  res.json({ boardArr, currentPlayer, currentColumns }); // sending boardArr, currentPlayer and currentColumns variables to client
+  res.json({
+    boardArr,
+    currentPlayer,
+    currentColumns,
+  }); // sending boardArr, currentPlayer and currentColumns variables to client
+});
+
+app.post("/newGame", async (req, res) => {
+  const newGameData = req.body;
+
+  const resetCurrentColumns = newGameData["resetCurrentColumns"];
+
+  currentColumns = [...resetCurrentColumns];
+
+  const resetGameBoard = newGameData["newBoard"];
+  boardArr = [...resetGameBoard];
+
+  const player = newGameData["player"];
+
+  currentPlayer = player;
```

Comment:
This overall change is good, except this part. You shouldn't accept the entire game state blindly from the user. Suppose they send you a game state that is an array with 1,000,000 rows and a 1,000,000 columns to see if they can crash your server?

Reply:
I'll update it. Thank you.